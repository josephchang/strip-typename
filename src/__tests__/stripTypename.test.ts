import stripTypename from '../';

describe('stripTypename', () => {
  describe('Given an Apollo GraphQL object with __typename', () => {
    let obj: any;

    beforeEach(() => {
      obj = {
        arrayItems: [
          {
            id: 1,
            colour: 'Red',
            nestedObj: {
              address: {
                line1: '742 Evergreen Terrace',
                city: 'Springfield',
                state: null,
                __typename: 'Address',
              },
              __typename: 'NestedObj',
            },
            __typename: 'ArrayItem',
          },
          {
            id: 2,
            colour: 'Yellow',
            nestedObj: {
              address: {
                line1: '31 Spooner Street',
                city: 'Quahog',
                state: 'Rhode Island',
                __typename: 'Address',
              },
              __typename: 'NestedObj',
            },
            __typename: 'ArrayItem',
          },
        ],
        obj: {
          integer: 1,
          primitive: true,
          nestedArrayItems: [
            {
              name: 'Nestle Toll House',
              __typename: 'NestedArrayItem',
            },
            {
              name: 'Conde Nest',
              __typename: 'NestedArrayItem',
            },
          ],
          __typename: 'Object',
        },
      };
    });

    describe('When the __typename is stripped', () => {
      let strippedObj: any;

      beforeEach(() => {
        strippedObj = stripTypename(obj);
      });

      it('should not have any __typename properties', () => {
        expect(strippedObj).toMatchSnapshot();
      });
    });
  });
});
