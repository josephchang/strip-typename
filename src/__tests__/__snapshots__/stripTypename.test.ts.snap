// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stripTypename Given an Apollo GraphQL object with __typename When the __typename is stripped should not have any __typename properties 1`] = `
Object {
  "arrayItems": Array [
    Object {
      "colour": "Red",
      "id": 1,
      "nestedObj": Object {
        "address": Object {
          "city": "Springfield",
          "line1": "742 Evergreen Terrace",
          "state": null,
        },
      },
    },
    Object {
      "colour": "Yellow",
      "id": 2,
      "nestedObj": Object {
        "address": Object {
          "city": "Quahog",
          "line1": "31 Spooner Street",
          "state": "Rhode Island",
        },
      },
    },
  ],
  "obj": Object {
    "integer": 1,
    "nestedArrayItems": Array [
      Object {
        "name": "Nestle Toll House",
      },
      Object {
        "name": "Conde Nest",
      },
    ],
    "primitive": true,
  },
}
`;
